project/src/main/java/com/google/autograder/servlets/helpers/API.java

```diff
@@ -68,6 +71,27 @@ public static String getCurrentUserAPIAuthorization() {
         return authorization;
     }
 
+    public static HttpURLConnection getAuthenticatedRequest(String method, String endpoint) {
+        try {
+            String authorization = getCurrentUserAPIAuthorization();
+
+            if (authorization == null) {
+                return null;
+            }
+
+            HttpURLConnection connection = (HttpURLConnection) new URL(endpoint).openConnection();
+
+            connection.setRequestMethod(method);
+            connection.setRequestProperty("Accept", "application/json");
+            connection.setRequestProperty("Authorization", authorization);
+
+            return connection;
+        } catch(Exception e) {
```

Comment:
Similar to other comments, prefer to catch the narrowest exception (probably IOException) and use the smallest possible try-catch block so that it's clear what can throw an exception and what can be thrown

Reply:
+1